index.js

```diff
@@ -0,0 +1,158 @@
+import * as core from "@actions/core";
+import { LinearClient } from "@linear/sdk";
+
+const ESCAPE = {
+  ">": "&gt;",
+  "<": "&lt;",
+  "&": "&amp;",
+  "\r\n": " ",
+  "\n": " ",
+  "\r": " ",
+};
+const ESPACE_REGEX = new RegExp(Object.keys(ESCAPE).join("|"), "gi");
+
+const getBugs = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [{ name: { eq: "Bug" } }, label ? { name: { eq: label } } : {}],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getChores = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [{ name: { eq: "Chore" } }, label ? { name: { eq: label } } : {}],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getFastTracks = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [
+          { every: { name: { nin: ["Bug", "Chore"] } } },
+          label ? { name: { eq: label } } : {},
+        ],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getProjects = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      project: { null: false },
+      state: { id: { in: stateIds } },
+      labels: label ? { name: { eq: label } } : {},
+    },
+  });
+  const projects = {};
+
+  for (let i = 0; i < issues.nodes.length; i++) {
+    const issue = issues.nodes[i];
+    if (!(issue._project.id in projects)) {
+      projects[issue._project.id] = await issue.project;
+    }
+  }
```

Comment:
```suggestion
  issues.nodes.forEach((issue) => {
    if (!(issue._project.id in projects)) {
      projects[issue._project.id] = await issue.project;
    }
  });
```

Reply:
This won't work. The inner function needs to be async and will not execute before the code that's below this section.

Reply:
ah ok. then you could probably use:

```
for (const issue of issues.nodes) {
  ...
}
```

Reply:
👌🏻 I've applied it on main directly.